File: src-tauri/src/commands/profiles.rs
use crate::core::types::{
    ApplyProfileRequest, ApplyProfileResult, ClearEnvironmentVariablesRequest,
    ClearEnvironmentVariablesResult, DeleteProfileDraftRequest, DuplicateProfileDraftRequest,
    PreviewProfileApplyRequest, PreviewProfileApplyResult, PreviewProfileWriteRequest,
    PreviewProfileWriteResult, ProfileDraft, ProfileSummary, ReorderProfileDraftsRequest,
    SaveProfileDraftRequest, StartCodexOAuthLoginResult, SwitchActiveProfileRequest,
    TestProfileConnectionRequest, TestProfileConnectionResult, UpdateProfileDraftRequest,
};
use crate::core::{env_health, profile};

#[tauri::command]
pub fn load_profile_summary() -> Result<ProfileSummary, String> {
    profile::load_profile_summary().map_err(|err| err.to_string())
}

#[tauri::command]
pub fn start_codex_oauth_login() -> Result<StartCodexOAuthLoginResult, String> {
    profile::start_codex_oauth_login().map_err(|err| err.to_string())
}

#[tauri::command]
pub fn save_profile_draft(request: SaveProfileDraftRequest) -> Result<ProfileDraft, String> {
    profile::save_profile_draft(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn update_profile_draft(request: UpdateProfileDraftRequest) -> Result<ProfileDraft, String> {
    profile::update_profile_draft(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn duplicate_profile_draft(
    request: DuplicateProfileDraftRequest,
) -> Result<ProfileDraft, String> {
    profile::duplicate_profile_draft(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn delete_profile_draft(request: DeleteProfileDraftRequest) -> Result<ProfileSummary, String> {
    profile::delete_profile_draft(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn reorder_profile_drafts(
    request: ReorderProfileDraftsRequest,
) -> Result<ProfileSummary, String> {
    profile::reorder_profile_drafts(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn preview_profile_write(
    request: PreviewProfileWriteRequest,
) -> Result<PreviewProfileWriteResult, String> {
    profile::preview_profile_write(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn preview_profile_apply(
    request: PreviewProfileApplyRequest,
) -> Result<PreviewProfileApplyResult, String> {
    profile::preview_profile_apply(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn apply_profile(request: ApplyProfileRequest) -> Result<ApplyProfileResult, String> {
    profile::apply_profile(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn test_profile_connection(
    request: TestProfileConnectionRequest,
) -> Result<TestProfileConnectionResult, String> {
    profile::test_profile_connection(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn switch_active_profile(
    request: SwitchActiveProfileRequest,
) -> Result<ProfileSummary, String> {
    profile::switch_active_profile(request).map_err(|err| err.to_string())
}

#[tauri::command]
pub fn clear_claude_environment_variables(
    request: ClearEnvironmentVariablesRequest,
) -> Result<ClearEnvironmentVariablesResult, String> {
    env_health::clear_environment_variables(request).map_err(|err| err.to_string())
}

File: src/lib/api/tauri/profiles.ts
import type {
  ApplyProfileRequest,
  ApplyProfileResult,
  DeleteProfileDraftRequest,
  DuplicateProfileDraftRequest,
  ListProfileModelsRequest,
  ListProfileModelsResult,
  PreviewProfileApplyRequest,
  PreviewProfileApplyResult,
  PreviewProfileWriteRequest,
  PreviewProfileWriteResult,
  ProfileDraft,
  ProfileSummary,
  ReorderProfileDraftsRequest,
  SaveProfileDraftRequest,
  StartCodexOAuthLoginResult,
  TestProfileConnectionRequest,
  TestProfileConnectionResult,
  UpdateProfileDraftRequest,
  UsageQueryResult,
  UsageScriptSaveRequest,
  UsageScriptState
} from "../../../types";
import type { RuntimeAdapter } from "../runtime";
import type { ProfileAdapter } from "../profiles";

export function tauriProfileAdapter(runtime: RuntimeAdapter): ProfileAdapter {
  return {
    ensureAppDirs: () => runtime.invoke<ProfileSummary>("ensure_app_dirs"),
    loadSummary: () => runtime.invoke<ProfileSummary>("load_profile_summary"),
    testConnection: (request: TestProfileConnectionRequest) =>
      runtime.invoke<TestProfileConnectionResult>("test_profile_connection", { request }),
    listModels: (request: ListProfileModelsRequest) =>
      runtime.invoke<ListProfileModelsResult>("list_profile_models", { request }),
    save: (request: SaveProfileDraftRequest) =>
      runtime.invoke<ProfileDraft>("save_profile_draft", { request }),
    startCodexOAuthLogin: () => runtime.invoke<StartCodexOAuthLoginResult>("start_codex_oauth_login"),
    update: (request: UpdateProfileDraftRequest) =>
      runtime.invoke<ProfileDraft>("update_profile_draft", { request }),
    duplicate: (request: DuplicateProfileDraftRequest) =>
      runtime.invoke<ProfileDraft>("duplicate_profile_draft", { request }),
    delete: (request: DeleteProfileDraftRequest) =>
      runtime.invoke<ProfileSummary>("delete_profile_draft", { request }),
    reorder: (request: ReorderProfileDraftsRequest) =>
      runtime.invoke<ProfileSummary>("reorder_profile_drafts", { request }),
    loadUsage: (profileId: string) =>
      runtime.invoke<UsageScriptState>("load_usage_script_state", { profileId }),
    saveUsage: (request: UsageScriptSaveRequest) =>
      runtime.invoke<UsageScriptState>("save_usage_script", { request }),
    testUsage: (request: UsageScriptSaveRequest) =>
      runtime.invoke<UsageQueryResult>("test_usage_script", { request }),
    queryUsage: (profileId: string) =>
      runtime.invoke<UsageQueryResult>("query_profile_usage", { profileId }),
    deleteUsage: (profileId: string) =>
      runtime.invoke<UsageScriptState>("delete_usage_script", { profileId }),
    previewWrite: (request: PreviewProfileWriteRequest) =>
      runtime.invoke<PreviewProfileWriteResult>("preview_profile_write", { request }),
    previewApply: (request: PreviewProfileApplyRequest) =>
      runtime.invoke<PreviewProfileApplyResult>("preview_profile_apply", { request }),
    apply: (request: ApplyProfileRequest) =>
      runtime.invoke<ApplyProfileResult>("apply_profile", { request })
  };
}
